R/bcdc_search.R

```diff
@@ -177,8 +181,8 @@ bcdc_search <- function(..., license_id = NULL,
   })
 
   query <- paste0(
-    terms, ifelse(nzchar(terms), "+", ""),
-    paste(names(facets), facets, sep = ":", collapse = "+"))
+    terms, ifelse(length(facets) > 0, "+", ""),
+    ifelse(length(facets) > 0, paste(names(facets), paste0("\"", facets, "\""), sep = ":", collapse = "+"), ""))
```

Comment:
We don't _have_ to do this but to make this a bit more readable I wonder if using `glue` might make this a little clearer. Another idea would be to extract this to a function and write a test to make sure it is doing what we want. I find it hard to grok exactly what this is doing. 

We also might want to implement our own `is_empty` function to avoid duplicating this `length(facets) > 0`. 

Whatever the case, a comment I think should be added to explain what's happening here.

Reply:
Comment added, as well as an `is_list_empty()` fxn to utils.  It took me awhile to grok this code to debug, but I didn't keep that experience in my head long enough to add "general code improvement" to dod for the next dev. Gr8 suggestions.